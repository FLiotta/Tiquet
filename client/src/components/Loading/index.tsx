// Packages
import React from 'react';
import cn from 'classnames';

// Project
import Spinner from '../../assets/images/spinner.svg';
import './styles.scss'

interface LoadingProps {
  display?: Boolean
};

const Loading = ({ display }: LoadingProps): JSX.Element => (
  <div className={cn('loading', {
    'loading--hidden': !display
  })}>
    <img src={Spinner} />
  </div>
);

export default Loading;