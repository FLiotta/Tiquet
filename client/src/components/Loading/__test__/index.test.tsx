// Packages
import * as React from 'react';
import { render } from '@testing-library/react';

// Project
import Loading from '../index';


describe('<Loading />', () => {
  it('Matches snapshot', () => {
    const { container } = render(
      <Loading display />
    );

    expect(container.firstChild).toMatchSnapshot();
  });

  it('Not visible when false display prop', () => {
    const { container } = render(
      <Loading />
    );
    expect(container.firstChild).toHaveClass('loading--hidden');
  });
});