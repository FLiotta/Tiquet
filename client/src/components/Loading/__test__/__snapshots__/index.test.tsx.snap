// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Loading /> Matches snapshot 1`] = `
<div
  class="loading"
>
  <img
    src="[object Object]"
  />
</div>
`;
